app/assets/javascripts/dashboard.coffee

```diff
@@ -3,5 +3,5 @@
 # You can use CoffeeScript in this file: http://coffeescript.org/
 
 $ ->
-  $("select#dashboard_filter_interval").on "change", (e) ->
+  $("#filters.submit-on-change select#filters_interval").on "change", (e) ->
```

Comment:
Since HTML id tags are supposed to be globally unique on the page, only the `select#filters_interval` is needed here, though perhaps the rest is good for self-documentation.

Reply:
So the reason for this is because this jquery will run on *every* page, and we use `#filters` on other pages, and we're adding something like `interval` ranges to other pages but they don't need the auto-submit logic